### PR TITLE
Support email_user_id

### PR DESCRIPTION
**Pull request by rachaelcodes**
Adds a MOCK_SSO_EMAIL_USER_ID setting, and updates API responses to return `email_user_id`:
- `/o/introspect/`
- `/api/v1/user/me`
- `/api/v1/user/search/`
- `/api/v1/user/introspect/`, which can use the email_user_id for searching

Also updates README with previously missing endpoints.

### CODE REVIEW

File: .sample.env [hunk 1]
```diff
@@ -1,3 +1,4 @@
 MOCK_SSO_PORT=
 MOCK_SSO_USERNAME=
+MOCK_SSO_EMAIL_USER_ID=
```
Comment (rachaelcodes) [inline .sample.env:3]: Adding this was requested in the ticket - as it is only returned as-is by the `/o/introspection` call, is a user-specified `email_user_id` needed (I'm not fully sure how the mock api is being used), or would a generic mock email user id work?

File: sample.env [hunk 2]
```diff
@@ -1,6 +1,8 @@
 MOCK_SSO_SCOPE=data-hub:internal-front-end
 # Set the username that you want to use with this mock SSO service
 MOCK_SSO_USERNAME=CHANGEME
+# Set the email_user_id that you want to use with this service
+MOCK_SSO_EMAIL_USER_ID=CHANGEME
```
Comment (reupen) [inline sample.env:5]: Would be good to use a more realistic example that's in the form of an email address (so e.g. `change-me@id.mock-sso`).

File: app/oauth/introspect.js [hunk 3]
```diff
@@ -9,6 +9,7 @@ const introspect = (scope, username) => {
       active: true,
       exp: 2524608000,
       username,
+      emailUserID,
```
Comment (reupen) [inline app/oauth/introspect.js:12]: Will this result in the property name being `emailUserID` – should it be `email_user_id: emailUserID`?

**Comment by rachaelcodes** [comment]
> Does `tests/unit/oauth/introspect.test.js` need an update?

Yes it does, thanks @reupen  - I've added this now.

File: app/api/user-list.json [hunk 4]
```diff
@@ -1,1808 +1,2109 @@
...
+     "email_user_id": "Corbin93-93098579@id.mock-sso"
+   },
+   {
      "user_id": "a9355c1f-7d3a-4894-a8bc-9d5ad149e23d",
      "first_name": "Johnny",
      "last_name": "Wehner",
-     "email": "Sylvan.Gerhold23@hotmail.com"
-  },
-  {
+     "email": "Sylvan.Gerhold23@hotmail.com",
```
Comment (elcct) [inline app/api/user-list.json:41]: Should we use test emails that look like real? Maybe we could at least change `hotmail.com` to something that doesn't exist.

Comment (rachaelcodes) [reply app/api/user-list.json]: Sounds fair - have updated this.

**Comment by elcct** [comment]
Looks like something is up with CicleCI config.

**Pull request merged**